### PR TITLE
Add module for twitter typeahead.js.

### PR DESCRIPTION
**Pull request by dw9**
Adds a new GWT module for twitter's typeahead.js [1] widget. Please also see the corresponding pull request in gwtbootstrap3-demo:

https://github.com/gwtbootstrap3/gwtbootstrap3-demo/pull/68

[1] https://github.com/twitter/typeahead.js/

### CODE REVIEW

File: src/main/java/org/gwtbootstrap3/extras/typeahead/client/TypeaheadClientBundle.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.google.gwt.resources.client.ClientBundle;
+import com.google.gwt.resources.client.TextResource;
+
+/**
+ * @author Florian Kremser <florian.kremser@sage.com>
+ */
+public interface TypeaheadClientBundle extends ClientBundle {
+  static final TypeaheadClientBundle INSTANCE = GWT.create(TypeaheadClientBundle.class);
+
+  @ClientBundle.Source("resource/js/typeahead.jquery.min.js")
```
Comment (jgodi) [inline src/main/java/org/gwtbootstrap3/extras/typeahead/client/TypeaheadClientBundle.java]: Rename file to typeahead.jquery-VERSION.min.cache.js to make the version apparent and leverage gwt caching. Please.

File: src/main/resources/org/gwtbootstrap3/extras/typeahead/client/resource/css/typeahead.css [hunk 2]
```diff
@@ -0,0 +1,86 @@
+/*
```
Comment (jgodi) [inline src/main/resources/org/gwtbootstrap3/extras/typeahead/client/resource/css/typeahead.css]: Same here please. typeahead-VERSION.min.cache.css. (also minifiy this)

**Comment by jgodi** [comment]
From initial glance this looks good. I will take a more in-depth look at this later and let some more eyes get on this. Funny, we just got the SuggestBox wrapped too, so this will give people another options.

@ArloL mind taking a look at this?

Thanks!

**Comment by dw9** [comment]
Ok, I've made the requested changes.

**Comment by jgodi** [comment]
Looks good, thanks man. We will many options for users now.

**Pull request merged**